integration-tests/src/lottery.rs

```diff
@@ -0,0 +1,255 @@
+use core::panic;
+
+use crate::*;
+
+#[test]
+fn can_set_and_accrue_lottery_prize_split() {
+	ExtBuilder::default().build().execute_with(|| {
+		// first round with default prize split.
+		assert_ok!(Lottery::buy_ticket(Alice.sign(), 10));
+		assert_ok!(Lottery::buy_ticket(Bob.sign(), 20));
+		assert_ok!(Lottery::buy_ticket(Charlie.sign(), 30));
+		assert_ok!(Lottery::buy_ticket(Dave.sign(), 40));
+
+		// Verify
+		System::assert_has_event(RuntimeEvent::Lottery(
+			pallet_lottery::Event::<Runtime>::TicketsBought {
+				id: Alice.account(),
+				number: 10u32,
+				total_price: 10 * DOLLAR,
+			},
+		));
+
+		assert_balance(Alice.account(), INITIAL_BALANCE - 10 * DOLLAR);
+		assert_balance(Bob.account(), INITIAL_BALANCE - 20 * DOLLAR);
+		assert_balance(Charlie.account(), INITIAL_BALANCE - 30 * DOLLAR);
+		assert_balance(Dave.account(), INITIAL_BALANCE - 40 * DOLLAR);
+
+		//PrizePoolAccount: AccountId = AccountId::from([0xFF; 32]);
+		assert_balance(PrizePool.account(), 100 * DOLLAR);
+
+		// LotteryPayoutPeriod is one DAY
+		Lottery::on_finalize(DAY);
+
+		if let RuntimeEvent::Lottery(pallet_lottery::Event::<Runtime>::LotteryWon {
+			user,
+			won_fund,
+			tax,
+		}) = System::events().into_iter().last().unwrap().event
+		{
+			println!("{:?} won. Amount: {}, tax: {}", user, won_fund, tax);
+			assert_balance(user, INITIAL_BALANCE - 40 * DOLLAR + won_fund);
+			assert_balance(Treasury.account(), tax);
+			assert_balance(PrizePool.account(), 0);
+		} else {
+			panic!("Last event is not LotteryWon. Events: \n{:?}", System::events());
+		}
+
+		// Governance set prize split.
+		let prize_split = vec![Percent::from_percent(80), Percent::from_percent(20)];
+		let call = Box::new(RuntimeCall::Lottery(pallet_lottery::Call::set_prize_split {
```

Comment:
I think we can add a utility function that automatically passes governance call
fn dispatch_governance_call(call: RuntimeCall) {
    // Obtain the current governance authorities
    // Propose using the first member and vote using the rest
    // call finalize to dispatch the call
}